datafusion/core/src/physical_planner.rs

```diff
@@ -719,14 +720,16 @@ impl DefaultPhysicalPlanner {
                     }
 
                     let logical_input_schema = input.schema();
-                    let physical_input_schema = input_exec.schema();
+                    // Extend the schema to include window expression fields as builtin window functions derives its datatype from incoming schema
```

Comment:
Shouldn't `input.schema()` reflect all the columns that the input produces?

Or does the `WindowAggExec` create new columns "internally" by evaluating the window expressions?

Reply:
input schema is schema from previous plan node(no window expressions). afaik windows expression column being added separately

https://github.com/apache/arrow-datafusion/blob/0116e2a9b4a3ed4491802e19195769b96b7a971a/datafusion/expr/src/logical_plan/plan.rs#L2045